x-pack/plugins/lens/public/indexpattern_datasource/operations/layer_helpers.ts

```diff
@@ -30,6 +30,81 @@ interface ColumnChange {
   shouldResetLabel?: boolean;
 }
 
+interface ColumnCopy {
+  layer: IndexPatternLayer;
+  columnId: string;
+  sourceColumn: IndexPatternColumn;
+  sourceColumnId: string;
+  shouldDeleteSource?: boolean;
+  indexPattern: IndexPattern;
+}
+
+export function copyColumn({
+  layer,
+  columnId,
+  sourceColumn,
+  sourceColumnId,
+  shouldDeleteSource,
+  indexPattern,
+}: ColumnCopy): IndexPatternLayer {
+  let newColumns = {
+    ...layer.columns,
+    [columnId]: { ...sourceColumn },
+  };
+
+  const operationDefinition = operationDefinitionMap[sourceColumn.operationType];
+  const newIds: string[] = [];
+  if (operationDefinition.input === 'fullReference' && 'references' in sourceColumn) {
+    sourceColumn?.references.forEach((ref) => {
+      const newId = generateId();
+      newIds.push(newId);
+      const refId = Object.keys(layer.columns).find((c) => c === ref);
```

Comment:
Isn't each entry in `references` an ID? You can validate that the ID exists by using `layer.columns[ref]`, I don't think this loop is required.